src/components/MoneyTemporaryForRefactorRequestConfirmationList.js

```diff
@@ -723,6 +726,33 @@ function MoneyTemporaryForRefactorRequestConfirmationList({
                     error={merchantError ? translate('common.error.fieldRequired') : ''}
                 />
             )}
+            {isCategoryRequired && (
+                <MenuItemWithTopDescription
+                    shouldShowRightIcon={!isReadOnly}
+                    title={iouCategory}
+                    description={translate('common.category')}
+                    numberOfLinesTitle={2}
+                    onPress={() => Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_CATEGORY.getRoute(iouType, transaction.transactionID, reportID, Navigation.getActiveRouteWithoutParams()))}
+                    style={[styles.moneyRequestMenuItem]}
+                    titleStyle={styles.flex1}
+                    disabled={didConfirm}
+                    interactive={!isReadOnly}
+                    rightLabel={translate('common.required')}
+                />
+            )}
+            {isTagRequired && (
+                <MenuItemWithTopDescription
+                    shouldShowRightIcon={!isReadOnly}
+                    title={PolicyUtils.getCleanedTagName(iouTag)}
+                    description={policyTagListName}
+                    numberOfLinesTitle={2}
+                    onPress={() => Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_TAG.getRoute(iouType, transaction.transactionID, reportID, Navigation.getActiveRouteWithoutParams()))}
+                    style={[styles.moneyRequestMenuItem]}
+                    disabled={didConfirm}
+                    interactive={!isReadOnly}
+                    rightLabel={translate('common.required')}
+                />
+            )}
```

Comment:
Would you refactor this to reduce code duplication?

I'm thinking of something like this:

```tsx
    const classifiedMenuItems = [
        shouldShowSmartScanFields && {
            item: <MenuItemWithTopDescription
                shouldShowRightIcon={!isReadOnly && !isDistanceRequest}
                title={formattedAmount}
                description={translate('iou.amount')}
                interactive={!isReadOnly}
                onPress={() => {
                    if (isDistanceRequest) {
                        return;
                    }
                    if (isEditingSplitBill) {
                        Navigation.navigate(ROUTES.EDIT_SPLIT_BILL.getRoute(reportID, reportActionID, CONST.EDIT_REQUEST_FIELD.AMOUNT));
                        return;
                    }
                    Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_AMOUNT.getRoute(iouType, transaction.transactionID, reportID, Navigation.getActiveRouteWithoutParams()));
                }}
                style={[styles.moneyRequestMenuItem, styles.mt2]}
                titleStyle={styles.moneyRequestConfirmationAmount}
                disabled={didConfirm}
                brickRoadIndicator={shouldDisplayFieldError && TransactionUtils.isAmountMissing(transaction) ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : ''}
                error={shouldDisplayFieldError && TransactionUtils.isAmountMissing(transaction) ? translate('common.error.enterAmount') : ''}
            />,
            isRelevant: true,
        },
        {
            item: <MenuItemWithTopDescription
                shouldShowRightIcon={!isReadOnly}
                shouldParseTitle
                title={iouComment}
                description={translate('common.description')}
                onPress={() => {
                    if (isEditingSplitBill) {
                        Navigation.navigate(ROUTES.EDIT_SPLIT_BILL.getRoute(reportID, reportActionID, CONST.EDIT_REQUEST_FIELD.DESCRIPTION));
                        return;
                    }
                    Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_DESCRIPTION.getRoute(iouType, transaction.transactionID, reportID, Navigation.getActiveRouteWithoutParams()));
                }}
                style={[styles.moneyRequestMenuItem]}
                titleStyle={styles.flex1}
                disabled={didConfirm}
                interactive={!isReadOnly}
                numberOfLinesTitle={2}
            />,
            isRelevant: true,
        },
        {
            item: <MenuItemWithTopDescription
                shouldShowRightIcon={!isReadOnly}
                title={isMerchantEmpty ? '' : iouMerchant}
                description={translate('common.merchant')}
                style={[styles.moneyRequestMenuItem]}
                titleStyle={styles.flex1}
                onPress={() => {
                    if (isEditingSplitBill) {
                        Navigation.navigate(ROUTES.EDIT_SPLIT_BILL.getRoute(reportID, reportActionID, CONST.EDIT_REQUEST_FIELD.MERCHANT));
                        return;
                    }
                    Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_MERCHANT.getRoute(iouType, transaction.transactionID, reportID, Navigation.getActiveRouteWithoutParams()));
                }}
                disabled={didConfirm}
                interactive={!isReadOnly}
                brickRoadIndicator={merchantError ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : ''}
                error={merchantError ? translate('common.error.fieldRequired') : ''}
            />,
            isRelevant: isMerchantRequired,
        },
        // ...
    ];

    const relevantMenuItems = classifiedMenuItems
        .filter((classifiedMenuItem) => classifiedMenuItem.isRelevant)
        .map((classifiedMenuItem) => classifiedMenuItem.item);

    const supplementaryMenuItems = classifiedMenuItems
        .filter((classifiedMenuItem) => !classifiedMenuItem.isRelevant)
        .map((classifiedMenuItem) => classifiedMenuItem.item);
```

We would show `relevantMenuItems` above the "Show more" pill and the `supplementaryMenuItems` below that pill.